homeassistant/components/accuweather/const.py

```diff
@@ -61,255 +64,264 @@
     ATTR_CONDITION_WINDY: [32],
 }
 
-FORECAST_DAYS = [0, 1, 2, 3, 4]
-
-FORECAST_SENSOR_TYPES = {
+FORECAST_SENSOR_TYPES: Final[dict[str, SensorDescription]] = {
     "CloudCoverDay": {
-        ATTR_DEVICE_CLASS: None,
-        ATTR_ICON: "mdi:weather-cloudy",
-        ATTR_LABEL: "Cloud Cover Day",
-        ATTR_UNIT_METRIC: PERCENTAGE,
-        ATTR_UNIT_IMPERIAL: PERCENTAGE,
+        "device_class": None,
```

Comment:
Why did we remove the use of constants?

Reply:
Because they're not really needed with `TypedDict`. Its keys behave like constants. But they don't need to be imported and they bonded to corresponding `TypedDict`.

Reply:
constants help with auto completions and IDE's. I suggest on keeping them as much as possible.

Reply:
`mypy` returns an error when constants are used with `TypedDict`:
```
error: TypedDict key must be a string literal; expected one of ('device_class', 'icon', 'label', 'unit_metric', 'unit_imperial', ...)
```

Reply:
You need to declare them `Final` and it will work.

Reply:
If we type the constants as `Final` it should work.

Reply:
Didn't know that, I'll revert the constants, thanks.

Reply:
But first I have to add `Final` to the `homeassistant.const` for example for `ATTR_ICON`.

Reply:
Yes, it should be fine to do that in the same PR. Or create separate PR to add `Final` to all constants in `homeassistant.const`.

Reply:
Please take a look https://github.com/home-assistant/core/pull/50858